File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  digits = tostring(number)
  exponent = #digits
  sum = 0

  digits:gsub(".", function (digit)
    sum = sum + tonumber(digit)^exponent
  end)

  return number == sum
end

return ArmstrongNumbers

File: solutions/javascript/armstrong-numbers/1/armstrong-numbers.js
export const isArmstrongNumber = (candidate) => {
  const digits = candidate.toString();
  const count  = digits.length;

  let sum = 0;

  for (let i = 0; i < count; i++) {
    const digit = +digits[i];
    sum += digit ** count;
  }

  return sum === candidate;
};
